pkg/sinks/forward/forward.go

```diff
@@ -443,3 +551,20 @@ func errorArrayToMap(errs []error) map[string]int64 {
 	}
 	return result
 }
+
+// getBackOffConditions configures the retry backoff strategy based on whether its a fallbackSink or primary sink.
+func (df *DataForward) getBackOffConditions(isFallbackSink bool) (wait.Backoff, dfv1.OnFailureRetryStrategy) {
+	// If we want for isFallbackSink we will return an infinite retry which will keep retrying post exhaustion till it succeeds
+	if isFallbackSink {
+		return wait.Backoff{
+			Duration: dfv1.DefaultRetryInterval,
+			// Always account for the first try
+			Steps: dfv1.DefaultRetrySteps + 1,
+		}, dfv1.OnFailureRetry
+	}
+	// Initial interval duration and number of retries are taken from DataForward settings.
+	backoff := df.opts.retryStrategy.GetBackoff()
+	// Adding a step for the first try
+	backoff.Steps = backoff.Steps + 1
```

Comment:
If we need to do this trick, why don't we correct the definition in the spec?

Reply:
I had thought about it, just one corner case where the user gives 0 somehow in the spec. Then there will be no sink writes happening.
This just ensures that we try `atleast` once, regardless of what the user gives us. 
I couldn't think a valid case where the user would not want that.